include/cppkafka/buffer.h

```diff
@@ -81,6 +81,17 @@ class CPPKAFKA_API Buffer {
             throw Exception("Invalid buffer configuration");
         }
     }
+    
+    /**
+     * Constructs a buffer from two iterators in the range [first,last)
+     *
+     * \param first An iterator to the start of data
+     * \param last An iterator to the end of data (not included)
+     */
+    template <typename Iter>
+    Buffer(const Iter first, Iter last)
```

Comment:
Both of these can be `const`. I'd also add the same `static_assert` that the other constructors have, using `iterator_traits` to check what the contained type is.

Reply:
Nevermind, I didn't realize this was forwarding. Ignore my second comment.